sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/percentiles.scala

```diff
@@ -350,6 +352,36 @@ case class Median(child: Expression)
       newChild: Expression): Median = this.copy(child = newChild)
 }
 
+/**
+ * A factory used to create the inverse distribution function during the analysis phase.
+ * This factory is also an aggregate function that cannot be evaluated.
+ */
+trait InverseDistributionFactory extends AggregateFunction {
```

Comment:
can we move it to a new file?

Reply:
instead of having a factory, shall we create a fake `PercentileCont` first and then update its `left` expression later in `validateFunction`?

Reply:
e.g. we can put a null literal as a placeholder for `PercentileCont.left`, and replace it later

Reply:
We still need a base trait
```
trait SupportsOrderingWithinGroup { self: AggregateFunction =>
  def withOrderingWithinGroup(orderingWithGroup: Seq[SortOrder]): Expression
}

```

Reply:
if null literal is too hack, we can create a placeholder expression.

Reply:
I think using the factory model or using `SupportsOrderingWithinGroup` is just a difference in specific implementation methods, and I have no preference for the two.